linodecli/plugins/obj/objects.py

```diff
@@ -52,6 +52,13 @@ def upload_object(
         default=MULTIPART_UPLOAD_CHUNK_SIZE_DEFAULT,
         help="The size of file chunks when uploading large files, in MB.",
     )
+    parser.add_argument(
+        "--folder",
+        metavar="FOLDER",
+        type=str,
+        help="If set, the object will be uploaded to the "
+        "specified subdirectory.",
+    )
```

Comment:
I wonder if it'd make sense to have the directory be dynamically parsed from the bucket argument similar to how s3cmd does it:

```bash
# s3cmd put FILE [FILE...] s3://BUCKET[/PREFIX]
# linode-cli obj put FILE BUCKET[/PREFIX]

linode-cli obj put file.txt mybucket/path/to/subdirectory
```